File: app/lab1_20_.hs
import Data.List (sort)
import Data.List.Split (splitOn)
import Data.Char (ord)

main :: IO ()
main = do
    rawContent <- readFile "names.txt"
    let result =
            sum $
            zipWith ( \position name -> position * alphabeticalValue name) [1..]
            (sort $ splitOn "," (filter (/= '"') rawContent))
    print result


alphabeticalValue :: String -> Int
alphabeticalValue name = sum $ map charValue name
  where
    charValue char = ord char - ord 'A' + 1

File: app/lab1_20.py

def cound_alp_nums(filename='names.txt'):
    nsum = 0
    with open(filename) as f:
        names = sorted(map(lambda x: x.strip('"'), (','.join(f.readlines())).split(',')))
        for i, x in enumerate(names):
            nsum += sum(map(lambda y: ord(y) - ord('A') + 1, list(x))) * (i + 1)
    return nsum

if __name__ == '__main__':
    print(cound_alp_nums())